bin/pytorch_inference/Main.cc

```diff
@@ -105,6 +105,9 @@ bool handleRequest(ml::torch::CCommandParser::CRequestCacheInterface& cache,
                 } catch (std::runtime_error& e) {
                     resultWriter.writeError(request_.s_RequestId, e.what());
                     return std::nullopt;
+                } catch (...) {
```

Comment:
The idea was to have a catch-all for other errors related to inference and write them out using `resultWriter`.  From the client point (e.g. `evaluate.py`), it's very difficult to understand otherwise, why processor doesn't return any results.

Reply:
If an exception we had no idea existed happens then it's reasonable to terminate the program. At least that will mean that we find out what the exception was from the stack trace. If we ever saw a whole series of `Unknown Error` messages that wouldn't be very helpful in diagnosing the problem. Maybe what we should do in the long term is to propagate unknown exceptions to the main thread (using `exception_ptr` and `current_exception()`) and rethrow them there so they actually terminate the program. But let's not add that to this PR because it would be good to get it merged today to allow the follow-on work to progress. So for this PR just add a `TODO` to improve this in the future.